application/routes/web.php

```diff
@@ -13,18 +13,36 @@
 
 Route::get('/home', 'HomeController@index')->name('home');
 
+/* admin ログインページへのリダイレクト */
 Route::redirect('/', '/home');
 
+/* （ログインページ/ログインアクション）では、middlewareのguestを使用し、認証済みの場合にそのページへは遷移しないように制御している */
 Route::group(['prefix' => 'admin', 'namespace' => 'Admin', 'as' => 'admin.', 'middleware' => 'guest:admin'], function () {
     Route::get('login', 'LoginController@showLoginForm')->name('login');
     Route::post('login', 'LoginController@login')->name('login');
 });
 
+/* adminログイン済みのユーザーのlogout、/admin/homeへの遷移ではmidllewareのAuthで認証をかけている */
 Route::group(['prefix' => 'admin', 'namespace' => 'Admin', 'as' => 'admin.', 'middleware' => 'auth:admin'], function () {
     Route::post('logout', 'LoginController@logout')->name('logout');
     Route::get('home', 'HomeController@index')->name('home');
 });
 
+/* adminログイン後のリダイレクト */
 Route::redirect('/admin', '/admin/home');
 
+/* 管理側の機能をまとめる（ 例： prefix：viewのパスがadmin/products, namespace:controllerはAdmin/Productcontroller, as:admin.products.のように
+　　パスに名前をつけている） */
+Route::group(['prefix' => 'admin', 'namespace' => 'Admin', 'as' => 'admin.'], function(){
```

Comment:
これでグループ化できるよ
```suggestion
Route::group(['prefix' => 'admin', 'namespace' => 'Admin', 'as' => 'admin.', ‘middleware’ => ‘auth:admin’], function(){
```

Reply:
ヒューマンエラーを防ぐ為にもこっちの方がいいかと
```suggestion
Route::prefix('admin')->namespace('Admin')->middleware(['auth:admin'])->group(function () {
```

Reply:
この書き方だと、
->group(function(){
　この中に各routeを書いていくんですか？
}
また、'as' => 'admin' については、->as('admin')のようにチェーンで繋げていくんですか？
すみません、そもそもこの記法だとヒューマンエラーを防げるという点が分かっておりません。

Reply:
> ->group(function(){
>　この中に各routeを書いていくんですか？
> }

その認識で合ってます

> また、'as' => 'admin' については、->as('admin')のようにチェーンで繋げていくんですか？

例にasが漏れてました、すみません。
asも入れると↓のような感じになります。
```
Route::prefix('admin')->namespace('Admin')->as('admin')->middleware(['auth:admin'])->group(...
```

> すみません、そもそもこの記法だとヒューマンエラーを防げるという点が分かっておりません。

Phpstorm等のエディターで補完が効くので、タイポ等のヒューマンエラーを防げるという点と、単純に見やすいっていうだけです。

Reply:
なるほど、参考にします。